### ISSUE TITLE
Remove submissions.current_publication_id column (circular reference)

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#6174]
The `submissions` table contains a `current_publication_id` column, which is a foreign key to `publications` indicating the newest publication.

However, this creates a circular DB reference, as `publications` refers back to `submissions` (`submission_id`).

The `current_publication_id` can always be derived:
```
SELECT s.submission_id, p.publication_id AS current_publication_id
FROM submissions s
    JOIN publications p ON (p.submission_id = s.submission_id)
    LEFT JOIN publications p_nonexistent ON (p_nonexistent.submission_id = s.submission_id AND p_nonexistent.publication_id > p.publication_id)
WHERE p_nonexistent.submission_id IS NULL;
```
This is a little hairy to read but...
- Efficient to execute (MySQL recognizes this as a `Not exists` condition)
- Removes a potential source of data inconsistency
- Resolves the circular reference.

**Comment by asmecher** [comment]
@NateWr, I think we spoke about this before, and it probably needs more consideration before I get started. In particular...
- Would a relationship table be a better idea? https://dba.stackexchange.com/questions/102903/is-it-acceptable-to-have-circular-foreign-key-references-how-to-avoid-them

**Comment by NateWr** [comment]
>  Does `current_publication_id` always refer to the *newest* publication entry?

No. It gets the most recently published publication OR if there is no published publication the newest publication. 

The current publication determines which publication will be used when displaying information about a submission (eg - title, authors). So everywhere a specific publication is not requested.

> Would a relationship table be a better idea?

I'd definitely defer to your judgement on this. I didn't add a `is_current` column to `publications` because it's important that there is always a 1-to-1 relationship here. A submission should never have more than one current publication.

**Comment by asmecher** [comment]
Thanks, @NateWr, I'll give this some more thought.
